src/engine/controller/Controller.java

```diff
@@ -41,57 +59,56 @@ public ProtoUserDataStream receiveUserRequest(ProtoUserDataStream data) {
     	// don't pass the value 10, pass the value actually want
     	// also need logic to split this into many many requests
     	
+    	final int MAX_THREADS = 50;
+    	
     	if (data == null || data.getInput().size() < 1) {
     	    throw new IllegalArgumentException("Input data can't be null");
     	}
     	
     	Iterator<Integer> dataIt = data.getInput().iterator();
+    	ExecutorService threadPool = Executors.newFixedThreadPool(MAX_THREADS);
+    
+		
+		List<Future<?>> futures = new ArrayList<>();
+    	
     	while(dataIt.hasNext()) {
-    		ProtoComputeEngineDataStream individualStream = new ComputeEngineDataStream(dataIt.next());
-            ProtoComputeEngineDataStream returnData = sendComputeRequest(individualStream);
-            List<String> dataConv = new ArrayList<String>();
-            Float currentArea = returnData.getArea();
-            
-            try {
-            	String dataConvAdd = currentArea.toString();
-            	dataConv.add(dataConvAdd);
-            } catch (Throwable t) {
-            	System.out.println("Floating point data unable to convert to string for storage.");
-            	t.printStackTrace();
-            }
-            
-            if (dataConv.size() != 1) {
-            	throw new IllegalStateException("Output of compute engine is invalud size.");
-            }
-            ProtoDataStream toStore = new DataStream(dataConv);
-            
-            
-            sendDataStoreRequest(toStore);
-             //dataIt.next();
-            // send data storage request
+    		int nextData = dataIt.next();
+    		Callable<ProtoComputeEngineDataStream> dataOutput = () -> {
+        		ProtoComputeEngineDataStream individualStream = new ComputeEngineDataStream(nextData);
+    			return sendComputeRequest(individualStream);
+    		};
+    		futures.add(threadPool.submit(dataOutput));
     	}
     	
+
+    	
+    	futures.forEach(future -> {
+    		try {
+    			List<String> dataConv = new ArrayList<String>();
+    			Float currentArea = ((ProtoComputeEngineDataStream) future.get()).getArea();
```

Comment:
Why are there 2 try blocks. dataConv isn't in the same scope line 91 and 87

Reply:
Need two as we need to get future value as well as verify the conversion separately.,